src/platform_impl/macos/app_delegate.rs

```diff
@@ -36,10 +38,35 @@ lazy_static! {
             activation_hack::mouse_moved as extern "C" fn(&Object, Sel, id),
         );
 
+        decl.add_method(
+            sel!(application:openFile:),
+            application_open_file as extern "C" fn(&Object, Sel, id, id) -> BOOL,
+        );
+        decl.add_method(
+            sel!(application:openFiles:),
+            application_open_files as extern "C" fn(&Object, Sel, id, id),
+        );
+
         AppDelegateClass(decl.register())
     };
 }
 
+/// Copies the contents of the ns string into a `String` which gets returned.
+fn ns_string_to_rust(ns_string: id) -> String {
+    use cocoa::foundation::NSString;
+
+    let utf8_len = unsafe { ns_string.len() };
+    let utf8_ptr = unsafe { ns_string.UTF8String() } as *mut u8;
+    let utf8_slice = unsafe { std::slice::from_raw_parts(utf8_ptr, utf8_len) };
+    let mut utf8_vec = Vec::<u8>::with_capacity(utf8_len);
+    unsafe {
+        utf8_vec.set_len(utf8_len);
+    }
+    utf8_vec.copy_from_slice(utf8_slice);
+
+    unsafe { String::from_utf8_unchecked(utf8_vec) }
+}
```

Comment:
This seems like a very convoluted way to convert between a UTF8String and a Rust String. Chances are this also isn't the first time this had to be done in winit, so I **strongly** suggest to look at functions like `get_characters` or `insert_text`.